build/Makefile

```diff
@@ -491,22 +507,37 @@ build-allocator-image: $(ensure-build-image) build-allocator-binary build-licens
 	docker build $(agones_path)/cmd/allocator/ --tag=$(allocator_tag) $(DOCKER_BUILD_ARGS)
 
 ifeq ($(WITH_WINDOWS), 1)
+ifeq ($(WITH_ARM64), 1)
```

Comment:
Oh boy this is fun. I hope we don't have to include another architecture down the line 😆

Not for this PR, but I wonder if we could eventually do some kind of a list, so that you could do something more like make `make build ARCH=windows,arm64,amd64` rather than having all these `WITH_*` statements.